### PR TITLE
[iOS] Change the Fire Button and Fireproofing to eTLD+1

### PR DESCRIPTION
**Pull request by edulpn**
Task/Issue URL: https://app.asana.com/1/137249556945/task/1213690830833565
Tech Design URL: https://app.asana.com/1/137249556945/project/481882893211075/task/1213685386860317
CC: @samsymons @hassaanelgarem @dus7 @jaceklyp

### Description

Changes the Fire Button and Fireproofing to operate at the eTLD+1 domain level, matching macOS behavior. Fireproofing any subdomain now protects the entire registrable domain

### Testing Steps
1. Start from a `main` build installed 
2. Log into `old.reddit.com`
  3. Fireproof `old.reddit.com`
  4. Fire, verify you're still logged in
  5. Run the PR build
  6. Verify you're still logged in on `old.reddit.com (migration preserved fireproofing)
  7. Settings → Data Clearing → Fireproofed Sites: `reddit.com` should appear in the list (eTLD+1 normalized from `old.reddit.com`)
  8. Visit `reddit.com` and verify you're also logged in there
  9. Fire and verify you're still logged in on both `old.reddit.com` and `reddit.com`
  10. Go to Settings → Data Clearing → Fireproofed Sites, remove `reddit.com` from the list
  11. Visit `old.reddit.com` and `reddit.com` and verify you're logged out of both
  12. Change a DuckDuckGo search setting (e.g., theme), fire, verify the setting persists

### Impact and Risks
Medium: Could disrupt specific features or user flows

#### What could go wrong?
- Migration fails silently: Existing fireproofed domains don't carry over. Mitigated by migration pixels and unit tests covering normalization, deduplication, and PSL edge cases.
- eTLD+1 over-protection: Fireproofing `mysite.github.io` protects all of `github.io` (non-PSL shared hosting). This is a known, accepted tradeoff matching macOS behavior. The PSL covers major cases (github.io, blogspot.com, wordpress.com etc.).

### Notes to Reviewer

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Medium Risk**
> Changes core fireproofing and cookie-clearing behavior to operate on registrable domains (eTLD+1), with a launch-time migration and new matching logic; incorrect normalization or migration could cause unexpected login retention or data clearing across subdomains.
> 
> **Overview**
> Fireproofing and Fire Button data clearing now operate at the **eTLD+1 (registrable domain)** level: fireproofing any subdomain protects the whole site, and cookie removal/retention logic is updated to match cookies across sibling subdomains.
> 
> This introduces a new `fireproofingETLDPlus1` feature flag and a **launch-time migration** of existing fireproofed domains into a new normalized store, with pixels emitted for migration start/success/failure.
> 
> UI messaging is adjusted to display the normalized domain, tests/mocks are updated accordingly, and a new Maestro flow validates cross-subdomain cookie persistence and clearing.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1a3e6accfc6d07e5c810054bd988fa539615c12b. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: iOS/Core/Fireproofing.swift [hunk 1]
```diff
@@ -31,19 +33,28 @@ public protocol Fireproofing {
...
 
-    public init() {}
-
     @UserDefaultsWrapper(key: .fireproofingAllowedDomains, defaultValue: [])
-    private(set) public var allowedDomains: [String]
+    var legacyAllowedDomains: [String]
+
+    public var allowedDomains: [String] {
+        isFireproofingETLDPlus1Enabled() ? etldPlus1AllowedDomains : legacyAllowedDomains
+    }
```
[the remark this answers is not in the document]
Comment (brindy) [reply iOS/Core/Fireproofing.swift:59]: @edulpn this sounds legit.  Can probably add an explicit way to test this to the privacy test pages?

File: iOS/Core/WebCacheManager.swift [hunk 2]
```diff
@@ -233,6 +237,18 @@ public class WebCacheManager: WebsiteDataManaging {
 
 extension WebCacheManager {
 
+    private func shouldRemoveCookie(_ cookie: HTTPCookie, forDomains domains: [String], tld: TLD) -> Bool {
```
[the remark this answers is not in the document]
Comment (edulpn) [reply iOS/Core/WebCacheManager.swift:240]: `clear(dataStore:forDomains:)` handles per-domain/tab-scoped fire, not the Fire Button. The Fire Button uses `clear(dataStore:)` with `.all` scope, which clears everything except fireproofed domains and is working correctly with eTLD+1 matching. The per-domain fire path is out of scope for this project.
